ecs-logging-access/src/main/java/com/raynigon/ecs/logging/access/server/CustomTeeFilter.java

```diff
@@ -0,0 +1,115 @@
+package com.raynigon.ecs.logging.access.server;
+
+import ch.qos.logback.access.common.AccessConstants;
+import jakarta.servlet.*;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CustomTeeFilter implements Filter {
+    boolean active;
+
+    @Override
+    public void destroy() {
+        // NOP
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
+            throws IOException, ServletException {
+
+        if (active && request instanceof HttpServletRequest) {
+            try {
+                TeeHttpServletRequest teeRequest = new TeeHttpServletRequest((HttpServletRequest) request);
+                TeeHttpServletResponse teeResponse = new TeeHttpServletResponse((HttpServletResponse) response);
+
+                // System.out.println("BEFORE TeeFilter. filterChain.doFilter()");
+                filterChain.doFilter(teeRequest, teeResponse);
+                // System.out.println("AFTER TeeFilter. filterChain.doFilter()");
+
+                teeResponse.finish();
+                // let the output contents be available for later use by
+                // logback-access-logging
+                teeRequest.setAttribute(AccessConstants.LB_OUTPUT_BUFFER, teeResponse.getOutputBuffer());
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw e;
+            } catch (ServletException e) {
+                e.printStackTrace();
+                throw e;
+            }
+        } else {
+            filterChain.doFilter(request, response);
+        }
+
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        String includeListAsStr = filterConfig.getInitParameter(AccessConstants.TEE_FILTER_INCLUDES_PARAM);
+        String excludeListAsStr = filterConfig.getInitParameter(AccessConstants.TEE_FILTER_EXCLUDES_PARAM);
+        String localhostName = getLocalhostName();
+
+        active = computeActivation(localhostName, includeListAsStr, excludeListAsStr);
+        if (active)
+            System.out.println("TeeFilter will be ACTIVE on this host [" + localhostName + "]");
```

Comment:
Since this is a logging library, it seems reasonable to use a logger instead.

Reply:
Wont work, because its a logging library...
I will remove all the print statements.